mlflow/utils/process.py

```diff
@@ -54,6 +54,6 @@ def exec_cmd(
         exit_code = child.wait()
         if throw_on_error and exit_code != 0:
             raise ShellCommandException(
-                "Non-zero exit code: %s\n\nSTDOUT:\n%s\n\nSTDERR:%s" % (exit_code, stdout, stderr)
+                f"Non-zero exit code: {exit_code}\n\nSTDOUT:\n{stdout}\n\nSTDERR:\n{stderr}"
```

Comment:
Are `stdout` and `stderr` binary strings?

Reply:
No. because the argument `universal_newlines` is True. which ensures they are str type.

Reply:
Got it!